jupyterhub/tests/test_pages.py

```diff
@@ -1105,17 +1105,27 @@ async def test_bad_oauth_get(app, params):
     [
         (["users"], False),
         (["admin:users"], False),
-        (["users", "admin:users", "admin:servers"], True),
+        (["users", "admin:users", "admin:servers"], False),
```

Comment:
I have a question that might be stupid:

I understand why access to the admin ui through `admin-ui` shouldn't require `admin:users`, `admin:servers` scopes.
But shouldn't `admin:users`, `admin:servers` (which are the highest access permissions to servers and users, right?) give you access to the admin ui?

I guess what I'm trying to say, is should `"admin:users"`, `"admin:servers"` automatically enable the `admin-ui` scope?

Reply:
I think it's fine to keep the two concerns separate. One is about API/backend permissions (which is complicated since RBAC means there's no clear delineation of what an admin is), the other is about whether to activate the UI/frontend regardless of permissions. It also provides an easy way to disable the admin UI, perhaps if someone builds their own.

Reply:
Good question! @manics answer is what I'm thinking - permission to take actions via the API and permission to view the admin page can be kept separate. This makes it easy for a deployment to totally hide the admin ui, for example, even if some users technically have very high permissions.

Reply:
I think I understand the motivation of keeping this separate now, thank you both for taking the time to explain this.
Basically, the UI it's treated as its own resource, with its own set of permissions that shouldn't be implied from access to other resources, esp since access to those other resources makes sense regardless of the access to the UI.
Also, this new RBAC setup is super cool but also, super 🤯

Reply:
I just had a thought, is this considered a breaking change since people will need to add `admin-ui` to their existing scopes instead of relying on `[admin:users, admin:servers]`? I'm OK with releasing this in a minor release with instructions on adding the new scope, since it improves the usability of the admin/RBAC setup and it help with long term maintenance, but let's make it an explicit decision!

Reply:
Good point! I'm okay with this in a minor release as well, though I'm increasingly thinking the custom scopes / tokens-on-scopes should perhaps be called 3.0 anyway. Even though I'm trying hard not to break anything, it's a substantial enough change that a major release might communicate it better.